### PR TITLE
Fix regression preventing command palette from opening

### LINKED ISSUE #5887: Command Palette isn't opened after closing a workspace
### Description
 
Nothing happens when I'm trying to open Command Palette in Theia with no workspace is opened.

### Reproduction Steps

Make sure no workspace is opened and press `F1` to open Command Palette.

**OS and Theia version:**
Theia 0.9.0
Commit id 0953ba77fe0b9c90bd7c122355acedc177ea5d0b

**Diagnostics:** 

In browser's dev console:
```
Uncaught (in promise) TypeError: Cannot read property 'uri' of undefined
    at CppBuildConfigurationManagerImpl.push.../../packages/cpp/lib/browser/cpp-build-configurations.js.CppBuildConfigurationManagerImpl.getActiveConfig (cpp-build-configurations.ts:274)
    at Object.isVisible (cpp-build-configurations-ui.ts:211)
    at CommandRegistry.../../packages/core/lib/common/command.js.CommandRegistry.getVisibleHandler (command.ts:300)
    at CommandRegistry.../../packages/core/lib/common/command.js.CommandRegistry.isVisible (command.ts:255)
    at quick-command-service.ts:163
    at Array.filter (<anonymous>)
    at QuickCommandService.../../packages/core/lib/browser/quick-open/quick-command-service.js.QuickCommandService.normalize (quick-command-service.ts:163)
    at QuickCommandService.../../packages/core/lib/browser/quick-open/quick-command-service.js.QuickCommandService.getCommands (quick-command-service.ts:149)
    at QuickCommandService.../../packages/core/lib/browser/quick-open/quick-command-service.js.QuickCommandService.init (quick-command-service.ts:64)
    at PrefixQuickOpenService.<anonymous> (prefix-quick-open-service.ts:178)
```

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes #5887

- fixes issue where the command palette can not be opened when
there is no workspace set. (caused by changes made in the 'cpp-build-configuration#getActiveConfig')
- if no workspace is present, return `undefined` instead of attempting to get the activeConfig
for the workspace root.

#### How to test

1. with a closed workspace, check if the command palette can be opened (ex: <kbd>F1</kbd>)
2. with an opened workspace, check if the command palette can be opened (ex: <kbd>F1</kbd>)

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/cpp/src/browser/cpp-build-configurations.ts [hunk 1]
```diff
@@ -270,9 +270,13 @@ export class CppBuildConfigurationManagerImpl implements CppBuildConfigurationMa
     }
 
     getActiveConfig(root?: string): CppBuildConfiguration | undefined {
-        // Get the active workspace root for the given uri, else for the first workspace root.
-        const workspaceRoot = root ? root : this.workspaceService.tryGetRoots()[0].uri;
-        return this.activeConfigurations.get(workspaceRoot);
+        if (!root) {
```
Comment (akosyakov) [inline packages/cpp/src/browser/cpp-build-configurations.ts]: @vince-fugnitto should not you use `root` if it is not `undefined`?

I think the issue was that `this.workspaceService.tryGetRoots()[0]` returns `undefined` and we try to call `.uri` on it.

Comment (vince-fugnitto) [reply packages/cpp/src/browser/cpp-build-configurations.ts]: @akosyakov yes you're correct, I've updated the code again.

**Comment by marechal-p** [comment]
I fixed the other places where something like `.tryGetRoots()[0].uri` was done.

**Comment by marechal-p** [comment]
Fixed a last issue where code would throw but not stop any process on each workspace opening.

@akosyakov @vince-fugnitto @RomanNikitenko @azatsarynnyy sorry for the trouble.

**Pull request merged**